src/streamlink_cli/main.py

```diff
@@ -931,16 +931,11 @@ def main():
             # Close output
             if output:
                 output.close()
-
-            # Make sure current stream gets properly cleaned up
-            if stream_fd:
-                console.msg("Interrupted! Closing currently open stream...")
-                try:
-                    stream_fd.close()
-                except KeyboardInterrupt:
-                    sys.exit()
-            else:
-                console.msg("Interrupted! Exiting...")
+            console.msg("Interrupted! Exiting...")
+        finally:
+            if hasattr(stream_fd, "close"):
+                console.msg("Closing currently open stream...")
+                stream_fd.close()
```

Comment:
You'll need 
```py
try:
    stream_fd.close()
except KeyboardInterrupt:
    sys.exit()
```

Inside the finally block to replace the existing `stream_fd.close()`. It is assumed if the user is pressing ctrl-c they want the process to end, there may be other exceptions caused by that, but that is acceptable...